contentcuration/contentcuration/management/commands/mark_incomplete.py

```diff
@@ -0,0 +1,47 @@
+from django.core.management.base import BaseCommand
+from django.db.models import Exists
+from django.db.models import OuterRef
+from django.db.models import Q
+from le_utils.constants import content_kinds
+from le_utils.constants import exercises
+
+from contentcuration.models import AssessmentItem
+from contentcuration.models import ContentNode
+from contentcuration.models import File
+
+
+class Command(BaseCommand):
+    def handle(self, *args, **options):
+        exercise_check_query = AssessmentItem.objects.filter(contentnode=OuterRef('id')) \
+            .exclude(type=exercises.PERSEUS_QUESTION)\
+            .filter(
+                Q(question='') |
+                Q(answers='[]') |
+                (~Q(type=exercises.INPUT_QUESTION) & ~Q(answers__iregex='"correct":true'))  # hack to check if no correct answers
+            )
+        file_check_query = File.objects.filter(preset__supplementary=False, contentnode=OuterRef("id"))
+
+        invalid_nodes = ContentNode.objects.all().annotate(
+            has_files=Exists(file_check_query),
+            has_questions=Exists(AssessmentItem.objects.filter(contentnode=OuterRef("id"))),
+            invalid_exercise=Exists(exercise_check_query)
+        ).filter(
+            Q(title='') |
+            ~Q(kind_id=content_kinds.TOPIC) & (
+                (~Q(kind_id=content_kinds.EXERCISE) & Q(has_files=False)) |
+                Q(license=None) |
+                (Q(license__is_custom=True) & (Q(license_description=None) | Q(license_description=''))) |
+                (Q(license__copyright_holder_required=True) & (Q(copyright_holder=None) | Q(copyright_holder='')))
+            ) |
+            Q(kind_id=content_kinds.EXERCISE) & (
+                Q(has_questions=False) |
+                Q(invalid_exercise=True) |
+                Q(extra_fields__has_key='mastery_model') |
+                Q(extra_fields__mastery_model=exercises.M_OF_N) & (
+                    ~Q(extra_fields__has_key='m') | ~Q(extra_fields__has_key='n')
+                )
+            )
+        ).values_list('id', flat=True)
+
+        # Getting an error on bulk update with the annotations, so query again
```

Comment:
Just flagging that I don't get what might be a problem with bulk update in case someone else should have a look

Reply:
But doesn't seem as anything risky, right? Not sure about how performance works in such cases

Reply:
It should be better than looping through the items at least

Reply:
Okay and I actually didn't realize that we'll need to run this only once